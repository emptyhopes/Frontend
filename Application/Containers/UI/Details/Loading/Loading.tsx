import styled, { keyframes } from "styled-components";

import { PropsInterface } from "@/Application/Containers/UI/Details/Loading/Types/PropsInterface";

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

const StyledPage = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  width: 100vw;
  height: 100vh;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.3);

  & div {
    width: 75px;
    height: 75px;
    border: 5px solid transparent;
    border-color: #000000 transparent;
  }
`;

const StyledLoading = styled.div`
  width: 25px;
  height: 25px;
  border: 2px solid transparent;
  border-color: #000000 transparent;
  border-radius: 50%;
  animation: ${rotate} 1.5s linear infinite;
`;

const Loading: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  return (
    <>
      {props.page ? (
        <StyledPage>
          <StyledLoading />
        </StyledPage>
      ) : (
        <StyledLoading />
      )}
    </>
  );
};

export { Loading };
