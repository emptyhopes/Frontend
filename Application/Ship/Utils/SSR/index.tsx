import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
