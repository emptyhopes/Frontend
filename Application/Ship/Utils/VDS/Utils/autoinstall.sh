#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
