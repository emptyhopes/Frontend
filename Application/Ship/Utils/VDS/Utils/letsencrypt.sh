#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname