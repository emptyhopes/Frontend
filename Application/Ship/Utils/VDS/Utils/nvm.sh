#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
