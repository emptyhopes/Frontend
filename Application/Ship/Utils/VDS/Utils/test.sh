#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru