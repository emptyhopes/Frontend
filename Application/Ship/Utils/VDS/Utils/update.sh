#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes