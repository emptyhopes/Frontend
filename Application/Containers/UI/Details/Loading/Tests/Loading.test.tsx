import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Loading } from "@/Application/Containers/UI/Details/Loading/Loading";

test("Loading", () => {
  const { container } = RenderTests(<Loading />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
