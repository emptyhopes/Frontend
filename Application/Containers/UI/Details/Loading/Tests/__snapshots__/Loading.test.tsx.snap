// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading 1`] = `
.c0 {
  width: 25px;
  height: 25px;
  border: 2px solid transparent;
  border-color: #000000 transparent;
  border-radius: 50%;
  -webkit-animation: epmXAj 1.5s linear infinite;
  animation: epmXAj 1.5s linear infinite;
}

<div>
  <div
    class="c0"
  />
</div>
`;
